testdata/ecosystem/pipelines/nodejs.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: nodejs-ex-git
+spec:
+  params:
+    - name: APP_NAME
+      default: nodejs-ex-git
+    - name: GIT_REPO
+      default: 'https://github.com/digitalocean/sample-nodejs.git'
+    - name: GIT_REVISION
+      default: master
+    - name: IMAGE_NAME
+      default: "image-registry.openshift-image-registry.svc:5000/$(context.pipelineRun.namespace)/nodejs-ex-git"
+    - name: PATH_CONTEXT
+      default: .
+    - name: VERSION
+      default: latest
+  tasks:
+    - name: fetch-repository
+      params:
+        - name: URL
+          value: $(params.GIT_REPO)
+        - name: revision
+          value: $(params.GIT_REVISION)
+        - name: subdirectory
+          value: ''
+        - name: deleteExisting
+          value: 'true'
+      taskRef:
+        resolver: cluster
+        params:
+        - name: kind
+          value: task
+        - name: name
+          value: git-clone
+        - name: namespace
+          value: openshift-pipelines
+      workspaces:
+        - name: output
+          workspace: workspace
+    - name: build
+      params:
+        - name: IMAGE
+          value: $(params.IMAGE_NAME)
+        - name: TLSVERIFY
+          value: 'false'
+        - name: PATH_CONTEXT
+          value: $(params.PATH_CONTEXT)
+        - name: VERSION
+          value: $(params.VERSION)
+      runAfter:
+        - fetch-repository
+      taskRef:
+        resolver: cluster
+        params:
+        - name: kind
+          value: task
+        - name: name
+          value: s2i-nodejs
+        - name: namespace
+          value: openshift-pipelines
+      workspaces:
+        - name: source
+          workspace: workspace
+    - name: deploy
+      params:
+        - name: SCRIPT
+          value: oc rollout status dc/$(params.APP_NAME)
+      runAfter:
+        - build
+      taskRef:
+        resolver: cluster
+        params:
+        - name: kind
+          value: task
+        - name: name
+          value: openshift-client
+        - name: namespace
+          value: openshift-pipelines
+    - name: expose
+      params:
+        - name: SCRIPT
+          value: |
+            #!/bin/sh
+            oc expose dc/$(params.APP_NAME) --target-port=3000
+            oc expose svc $(params.APP_NAME) --target-port=3000
```

Comment:
1. Are both commands necessary?
2. It would be better to expose it in step, not in the pipeline. In real world you would only expose the app once, not on every application build

Reply:
Done